packages/select/src/vaadin-select.js

```diff
@@ -269,7 +310,7 @@ class Select extends DelegateFocusMixin(FieldMixin(SlotMixin(ElementMixin(Themab
       '_updateAriaExpanded(opened)',
       '_updateAriaRequired(required)',
       '_updateSelectedItem(value, _items, placeholder)',
-      '_rendererChanged(renderer, _overlayElement)'
+      '__rendererChanged(__renderer, _overlayElement)'
```

Comment:
Just wondering if having an observable property for `__renderer`, and all the other baggage that comes with it (e.g. `__initialized`) is really necessary.

Assuming your example:
```js
select.items = [
  { label: 'Jose', value: 'jose' },
  { component: 'hr' },
  { label: 'Pedro', value: 'pedro' },
  { label: 'Manolo', value: 'manolo', disabled: true }
];
```
Setting items should already be an observable change. 

Are there really cases where not setting any observable property should trigger the `__rendererChanged`? If not we could replace `__renderer` and `__computeRenderer` with a single getter.

Reply:
Or maybe the current assumption is that the user should set a `renderer`, and that's why the component assumes that `__rendererChanged` will be called during initialization. In that case we might want to adapt the initialization logic instead.

Reply:
I didn't probably understand you all the way. I've just realized that the `__initialized` property makes more sense in the grid case (where it actually came from to the select). In the grid case, there were some troubles with the prototype chain: when a property is defined as a dependency of an observer, Polymer saves the reference to the property defined in the class where the observer is defined and it becomes not possible to override it. As for the select, we could do the following instead:

```js
{
  __computedRenderer: {
    type: Function,
    computed: '__computeRenderer(renderer, __defaultRenderer)'
  }
}
```

Reply:
> Are there really cases where not setting any observable property should trigger the __rendererChanged

Yes. 

First, whenever the renderer changes, it is forwarded to the overlay in the `__rendererChanged` method.

When no custom renderer is provided initially, the default one is assumed to be used and passed to the overlay element. However, if having neither the `__initialized` property nor the `__defaultRenderer` property on the dependency list of `__computedRenderer`, Polymer will skip computing `__computedRenderer` because of all dependencies being undefined.

Reply:
After the internal discussion, we decided not to introduce compute-like methods as there is not that much sense in them for the select. The PR is updated.